script.js

```diff
@@ -30,3 +81,150 @@ menuLinks.forEach((link) => {
+  const span = document.createElement('hr');
+  span.setAttribute('class', 'line-span');
+  popBtns.setAttribute('class', 'pop-btn');
+  const liveBtn = document.createElement('button');
+  liveBtn.setAttribute('class', 'live-btn');
+  liveBtn.setAttribute('type', 'button');
+  liveBtn.innerHTML = 'See live <i class="fas fa-arrow-circle-right fa-lg"></i>';
+  const sourceBtn = document.createElement('button');
+  sourceBtn.setAttribute('class', 'source-btn');
+  sourceBtn.setAttribute('type', 'button');
+  sourceBtn.innerHTML = 'See Source <i class="fab fa-github fa-lg"></i>';
+
+  const closeBtnPop = document.createElement('i');
+  closeBtnPop.className = 'fas fa-times';
+  closeBtnPop.addEventListener('click', () => {
+    const popup = document.querySelector('.project-pop');
+    if (popup) {
+      popup.parentElement.removeChild(popup);
+    }
+  });
+
+  techPop.append(techPopType);
+  techPop.append(techPopSpan1);
+  techPop.append(techPopDevType);
+  techPop.append(techPopSpan2);
+  techPop.append(techPopDevYear);
+  ProjectPopImageDiv.append(ProjectPopImage);
+  usedLang.append(usedLangP1);
+  usedLang.append(usedLangP2);
+  usedLang.append(usedLangP3);
+  popBtns.append(liveBtn);
+  popBtns.append(sourceBtn);
+  container10.append(usedLang);
+  container10.append(span);
+  container10.append(popBtns);
+  projectPopContainer.append(closeBtnPop);
+  projectPopContainer.append(projectPopName);
+  projectPopContainer.append(techPop);
+  projectPopContainer.append(ProjectPopImageDiv);
+  projectPopContainer.append(ProjectPopDesc);
+  projectPopContainer.append(container10);
+  projectPop.append(projectPopContainer);
+  return projectPop;
+}
+const { body } = document;
+
+document.querySelectorAll('.pop-btn').forEach((btn) => {
+  btn.addEventListener('click', (event) => {
+    if (!document.querySelector('.project-pop')) {
+      const popupData = data[parseInt(event.target.id, 10)];
+      const popup = CreatePopup(popupData);
+      body.append(popup);
+    }
+  });
+});
+
+// ######## form client side validation ########
+
+const form = document.getElementById('form');
+const email = document.getElementById('email');
+const formBtn = document.querySelector('.submit-btn');
+
+function setErrorFor(input) {
+  const formControl = input.parentElement;
+  formControl.className = 'form-control error';
+}
+
+function setSuccessFor(input) {
+  const formControl = input.parentElement;
+  formControl.className = 'form-control success';
+}
+
+function isEmail(email) {
+  return /^[_a-z0-9-]+(\.[_a-z0-9-]+)*@[a-z0-9-]+(\.[a-z0-9-]+)*(\.[a-z]{2,4})$/.test(email);
+}
+
+function checkInputs() {
+  let errorMsg;
+  let valid = true;
+  const emailValue = email.value.trim();
+
+  if (emailValue === '') {
+    valid = false;
+    errorMsg = 'Email cannot be blank';
+    setErrorFor(email, 'Email cannot be blank');
+  } else if (!isEmail(emailValue)) {
+    valid = false;
+    errorMsg = 'Email is not valid';
+    setErrorFor(email, 'Email is not valid');
```

Comment:
- [ ] Great job implementing the `error message` 👍🏽 However, it is not displaying when an error occured. Kindly remedy that detail 🙏🏽 

It should display an error message about "Uppercase" here 👇🏽 

![Screenshot from 2023-07-28 01-01-28](https://github.com/gilberthappi/portfolio/assets/10439283/50c18bbe-fb0c-41fb-9feb-0ebb5efa1145)

It should display an error message concerning "blank email" here 👇🏽 

![Screenshot from 2023-07-28 01-07-03](https://github.com/gilberthappi/portfolio/assets/10439283/381d3f36-1f2b-4ba1-a6f2-350ddd3e7333)